front50-web/src/test/groovy/com/netflix/spinnaker/front50/controllers/NotificationControllerTck.groovy

```diff
@@ -188,15 +189,17 @@ class S3NotificationControllerTck extends NotificationControllerTck {
   def scheduler = Schedulers.from(Executors.newFixedThreadPool(1))
 
   @Shared
-  S3NotificationDAO s3NotificationDAO
+  NotificationDAO notificationDAO
 
   @Override
   NotificationDAO createNotificationDAO() {
     def amazonS3 = new AmazonS3Client(new ClientConfiguration())
     amazonS3.setEndpoint("http://127.0.0.1:9999")
     S3TestHelper.setupBucket(amazonS3, "front50")
 
-    s3NotificationDAO = new S3NotificationDAO(new ObjectMapper(), amazonS3, scheduler, 0, "front50", "test")
-    return s3NotificationDAO
+    def storageService = new S3StorageService(new ObjectMapper(), amazonS3, "front50", "test")
```

Comment:
I'm not sure what is going on here.
Is it possible/practical to test S3Storage service separate and write these tests in terms of the base controller?

Ideally this is base controller and there is another that tests StorageService where we can each plug our StorageService in and it verifies compliance (throw consistent exceptions, returns consistent results). I dont know how hard that is. i dont want to compromise your testing, but would like mine to benefit and ensure we stay in sync.